### DISCUSSION TITLE
[CLOSED] Use `get` for maps and `[..]` for lists?

### DISCUSSION THREAD

**Discussion by konnov** [informalsystems/quint#306] [Ideas]
Currently, we have the operator `f[e]` reserved for maps (TLA+ functions), and `l.nth(i)` reserved for lists.

I find that I always want to write `f.get(e)` knowing that `f` is a map (indeed, we can just write `f.get(e)` in Quint). Also, I would rather write `l[i]` for a list.

Shall we just use `l[i]` as a syntax sugar for lists instead of maps? We would still be able to use `get` and `nth` for maps and lists, respectively.

**Comment by shonfeder** [comment]
This would help with one of the inconsistencies i noted in #93, so I am in favor.

I think we should also consider renaming `map` to `dictionary` so we avoid all the ambiguities between the map data structure and the `map` operations over collections.

**Reply by konnov** [reply]
@shonfeder, you have envisioned all the problems with maps in May. I am still surprised how I am bumping my head against your list :D

**Reply by bugarela** [reply]
Alternatively, we could use words starting with capital letters for constructors and types, like `Map` and `Set`, and then have `map` for the operation over lists (or sets) and even `set` as a replacement for `with` or `update`.

To visualize and type, I really like how most of TNT is lowercase, but people are used to having capital letters in programming languages, so it's fine.

**Reply by konnov** [reply]
upper case `Map` and `Set` is an interesting idea

**Comment by konnov** [comment]
Implemented in #321. Closing the discussion.
